contracts/asset-proxy/contracts/src/bridges/DydxBridge.sol

```diff
@@ -0,0 +1,179 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "@0x/contracts-utils/contracts/src/Authorizable.sol";
+import "../interfaces/IERC20Bridge.sol";
+import "../interfaces/IDydxBridge.sol";
+import "../interfaces/IDydx.sol";
+
+
+contract DydxBridge is
+    IERC20Bridge,
+    IDydxBridge,
+    DeploymentConstants,
+    Authorizable
+{
+
+    /// @dev Callback for `IERC20Bridge`. Deposits or withdraws tokens from a dydx account.
+    ///      Notes:
+    ///         1. This bridge must be set as an operator of the dydx account that is being operated on.
+    ///         2. This function may only be called in the context of the 0x Exchange executing an order
+    ///            (ERC20Bridge is authorized).
+    ///         3. The order must be signed by the owner or an operator of the dydx account.
+    ///            This signature validated by the 0x Exchange.
+    ///         4. The `from` address must be the maker (and hence is the owner or operator of the dydx account).
+    ///            This is asserted during execution of this function.
+    /// @param from The sender of the tokens.
+    /// @param to The recipient of the tokens.
+    /// @param amount Minimum amount of `toTokenAddress` tokens to buy.
+    /// @param encodedBridgeData An abi-encoded `BridgeData` struct.
+    /// @return success The magic bytes if successful.
+    function bridgeTransferFrom(
+        address,
+        address from,
+        address to,
+        uint256 amount,
+        bytes calldata encodedBridgeData
+    )
+        external
+        onlyAuthorized
```

Comment:
I wonder if we should use `onlyAuthorized` here vs hard coding a `require(msg.sender == EXCHANGE)`. I don't think this contract needs to be upgradable or support multiple authorized addresses. There's probably an argument to be made for that, but my guess is this will be the first of many iterations before we want to worry about upgradability.

Reply:
Sure I'm down to nix the dependency and use something more lightweight for now.